src/session-pool.ts

```diff
@@ -952,56 +806,34 @@ export class SessionPool extends EventEmitter implements SessionPoolInterface {
   }
 
   /**
-   * Returns true if the pool has a session that is usable for the specified
-   * type, i.e. if a read-only session is requested, it returns true if the
-   * pool has a read-only or a read/write session. If a read/write session is
-   * requested, the method only returns true if the pool has a read/write
-   * session available.
-   * @param type The type of session.
+   * Returns true if the pool has a usable session.
    * @private
    */
-  _hasSessionUsableFor(type: types): boolean {
-    return (
-      this._inventory[type].length > 0 ||
-      this._inventory[types.ReadWrite].length > 0
-    );
+  _hasSessionUsableFor(): boolean {
+    return this._inventory.sessions.length > 0;
   }
 
   /**
-   * Attempts to get a session of a specific type. If the type is unavailable it
-   * may try to use a different type.
+   * Attempts to get a session.
    *
    * @private
    *
-   * @param {string} type The desired session type.
    * @param {number} startTime Timestamp to use when determining timeouts.
    * @returns {Promise<Session>}
    */
-  async _getSession(type: types, startTime: number): Promise<Session> {
-    if (this._hasSessionUsableFor(type)) {
-      return this._borrowNextAvailableSession(type);
-    }
-    // If a read/write session is requested and the pool has a read-only session
-    // available, we should return that session unless there is a session
-    // currently being prepared for read/write that is not already claimed by
-    // another requester.
-    if (
-      type === types.ReadWrite &&
-      this._hasSessionUsableFor(types.ReadOnly) &&
-      this.numWriteWaiters >= this.pendingPrepare
-    ) {
-      return this._borrowNextAvailableSession(type);
+  async _getSession(startTime: number): Promise<Session> {
+    if (this._hasSessionUsableFor()) {
+      return this._borrowNextAvailableSession();
     }
-
     if (this.isFull && this.options.fail!) {
       throw new SessionPoolExhaustedError(this._getLeaks());
     }
 
     let removeOnceCloseListener: Function;
     let removeListener: Function;
 
-    // Wait for the requested type of session to become available.
-    const availableEvent = type + '-available';
+    // Wait for a session to become available.
+    const availableEvent = 'available';
```

Comment:
We need to change the name of what is emitted here as 'available' is already an event caught in database: https://github.com/googleapis/nodejs-spanner/blob/1fadc69540015b696d678612e8aa370b13daa878/src/database.ts#L352
We can change it to 'session-available' here and also emit it at line number 1012: https://github.com/googleapis/nodejs-spanner/pull/1739/files#diff-d6f67152c7ddf6462713d84c5314cab643222c347a9b1c87f59498d8cd9c84abR1012
rather than 'readonly-available' and 'readwrite-available' as those are currently emitted an not caught anywhere

Reply:
I think @olavloite made a comment https://github.com/googleapis/nodejs-spanner/pull/1739#discussion_r1019983391 that even if we're not catching this `readonly-available` and `readwrite-available` in our codebase, it can be a breakable changes and it can affect our customers that rely on these events. 

we have some options:
1) remove `readonly-available` and `readwrite-available`
2) replace `readonly-available` and `readwrite-available` with `session-available`
3) add `session-available` (also mark `readonly-available` and `readwrite-available` as deprecated). 

Please confirm which option is the right to move forward, and I'm happy to update to code.

Reply:
@ko3a4ok I had a discussion with @olavloite and we should go ahead with option **3**. We do not want to remove `readonly-available` and `readwrite-available` as it might cause a breaking change for the customer.

Reply:
thanks for checking. Done

Reply:
@ko3a4ok We will also have to modify the event name like below 
```const availableEvent = 'session-available';```

Reply:
Thanks. done.